sqlmesh/core/renderer.py

```diff
@@ -168,6 +160,10 @@ def _render(
             jinja_env=jinja_env,
             schema=self.schema,
             runtime_stage=runtime_stage,
+            resolve_table=t.cast(
+                t.Callable[[str | exp.Expression], str],
+                jinja_env.globals["resolve_table"],
```

Comment:
Why are we doing this? I think we should just pass `self._resolve_table` directly

Reply:
Because the lambda in the jinja env closes around `table_mapping` as well which is not currently propagated to the macro evaluator. We do something similar for `resolve_tables=lambda e: ...` in the line below this.

Reply:
Yeah, my question was rather whether we should do the same lambda thing here instead of assuming that something is available in the jinja env.